src/server.c

```diff
@@ -5735,6 +5741,9 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
             "mem_cluster_links:%zu\r\n", mh->cluster_links,
             "mem_aof_buffer:%zu\r\n", mh->aof_buffer,
             "mem_allocator:%s\r\n", ZMALLOC_LIB,
+            "mem_overhead_hashtable_lut:%zu\r\n", server.overhead_hashtable_lut * sizeof(dictEntry*),
+            "mem_overhead_hashtable_rehashing:%zu\r\n", server.overhead_hashtable_rehashing * sizeof(dictEntry*),
```

Comment:
I would prefer just to keep this one. This one seems very actionable to me, since it indicates the temporary overhead for me. The other two seem more tightly coupled to the current implementation.

Reply:
`mem_overhead_hashtable_lut` is an important part of memory usage, we may regard it as "static" overhead, while the rehashing overhead is "dynamic".

When the overhead of rehashing grows large, `databases_rehashing_dict_count` can tell us the keys increasing happens in a few slots or many slots, which helps monitoring data flow.

These two may be not so intuitive but displaying them gives a more complete description of DB's hash tables.

Reply:
so you consider dictEntry part of the key in that respect? or is it also part of the "static" overhead?

Reply:
I don't mean any of these. I am just saying that both aspect of look up table's overhead should be displayed together.

Reply:
I tend to keep both. The overhead mainly consists of two parts:

1. the metadata of each key, which includes `dictEntry` and `robj`. The size of these metadata items is directly proportional to the number of keys, and they do not cause memory spikes due to rehashing.
2. the other part is the hash table or lookup table, which is related to the number of buckets. However, the number of buckets does not have a one-to-one relationship with the number of keys, and rehashing can cause memory spikes. Therefore, it is difficult to infer the actual size.

Hence, keeping both can better assist users in observing the memory usage related to the second point.

Reply:
i personally don't mind keeping both, we have other low level metrics already anyway.

p.s. till recently we had a `overhead.hashtable.slot-to-keys` in MEMORY STATS, which we now dropped.
we could have kept it and measure the memory wasted by 16k * sizeof(dict).

Reply:
A lot of my perspective comes from the idea that I don't like showing information to users unless it's clear what it's useful for. I think we as engineers come from a world where more data is better, but most people are not as technically sophisticated as us and we are asking them to parse more information. I want to keep Redis simple.

> These two may be not so intuitive but displaying them gives a more complete description of DB's hash tables.

I don't want end users to have a complete description, I want them to have a useful description. If you have a reason this information is useful, I am all for adding it.

Reply:
I suppose I'll also clarify I'm OK with `MEMORY STATS` having more detailed breakdown as well. We don't need all the information posted everywhere all the time.

Reply:
i don't like MEMORY STATS (i regret having it). many monitoring frameworks only collect data from INFO, so if a field is there, it can be easily collected, and if it's missing from INFO, it's not collected.
In my experience, that makes MEMORY STATS useful only for manual investigation with redis-cli.

p.s. we have another option (which i don't support) which is exposing it in INFO DEBUG (not officially part of the interface).

Reply:
i think that if we can't agree on everything, let's reduce the scope and merge the part we agree on.

Reply:
> In my experience, that makes MEMORY STATS useful only for manual investigation with redis-cli.

My understanding is that the two more detailed fields are basically **only** useful during manual investigations? From a monitoring perspective, you really just care that extra memory was being used during the rehashing.

Reply:
so `mem_overhead_hashtable_rehashing` for INFO, and the other two only in MEMORY?
let's do that.

Reply:
> so mem_overhead_hashtable_rehashing for INFO, and the other two only in MEMORY?
let's do that.

I am OK with this, since we can't agree on everything.